src/main/java/brave/opentracing/BraveSpanBuilder.java

```diff
@@ -88,6 +89,11 @@
     return withTag(key, value.toString());
   }
 
+  @Override public <T> BraveSpanBuilder withTag(Tag<T> tag, T value) {
+    new TagSetterHack<>(this, tag, value);
```

Comment:
Why can't you do `withTag(tag.getKey(), value)`? Is it because you don't know of which instance the value is? Could you do `instanceof` checks? Anyway, adding `void set(SpanBuilder s, T value);` to `Tag` sounds like a good solution. Maybe even something like `void set(Taggable s, T value);` where `Span` and `SpanBuilder` implement `Taggable`. (just brainstorming here)